backstory/src/main/java/com/google/sps/story/PromptManager.java

```diff
@@ -71,7 +77,7 @@ public void isTemplateRandomized(boolean isTemplateRandomized) {
    */
   public String generatePrompt() {
     // Prepare story-like prefix.
-    String prompt = "Once upon a time, ";
+    String prompt = "Once upon a time" + getFormattedLocation() + ", ";
```

Comment:
I feel like this method should be responsible for inserting spaces around the location text as needed. The `getFormattedLocation()` method shouldn't know how it's being used, and it not adding a prefixing space makes the method more flexible (you won't need to strip the space before using it in other places).

```
String locationString = getFormattedLocation();
if (locationString.length() > 0) {
  locationString = " " + locationString;
}
```

Reply:
That makes sense. I'll make this change. Thanks!

Reply:
Done